### PR TITLE
IFS-2502: Migrationsleitfaden ergaenzen HTTP-Invoker

### PR DESCRIPTION
**Pull request by FabianPerl**

### CODE REVIEW

File: isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc [hunk 1]
```diff
@@ -34,6 +34,15 @@ Spring HTTP-Invoker ist deprecated und wird ab sofort als Schnittstellenformat d
 Die Verwendung von REST-Schnittstellen wird im Baustein REST (siehe xref:isy-service-rest:konzept/master.adoc[] und xref:isy-service-rest:nutzungsvorgaben/master.adoc[]) erläutert.
 ====
 
+[WARNING]
+====
+Mit der Verwendung von `isy-security` in der IsyFact 3 wird standardmäßig anstelle von IsyFacts `AufrufKontext` Springs `SecurityContext` verwendet.
+
+Damit dennoch alte HTTP-Invoker Schnittstellen aufgerufen werden können (welche einen gefüllten AufrufKontext benötigen), wird eine spezielle `RemoteInvocationFactory` eingesetzt, welche Springs `SecurityContext` in ein `AufrufKontextTo` ad hoc per Request umwandelt.
```
Comment (kollmana) [inline isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc:41]: Does the application project need to implement or extend this special `RemoteInvocationFactory` or is it a ready-to-use part of the IsyFact Framework?

File: isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc [hunk 2]
```diff
@@ -34,6 +34,15 @@ Spring HTTP-Invoker ist deprecated und wird ab sofort als Schnittstellenformat d
...
+
+Weitere Informationen können unter
+xref:isy-serviceapi-core:nutzungsvorgaben/inhalt.adoc#kompatibilitaet_if1_2_3[Nutzungsvorgaben HTTP-Invoker] nachgelesen werden.
```
Comment (kollmana) [inline isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc:44]: Concerning the scenarios A) IF3 client calling IF1/2, and B) IF1/2 client calling IF3: What are the steps neccesary for migrating?

**Review by kollmana — COMMENTED** [review]
Does the application project need to implement or extend this special RemoteInvocationFactory or is it a ready-to-use part of the IsyFact Framework?

Concerning the scenarios A) IF3 client calling IF1/2, and B) IF1/2 client calling IF3: What are the steps neccesary for migrating?

File: isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc (hunk 1 above)
Comment (FabianPerl) [reply isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc:41]: As the [documentation for http invoker](https://isyfact.github.io/isyfact-standards-doku/latest/isy-serviceapi-core/nutzungsvorgaben/master.html#if3_client) describes, the `IsyHttpInvokerProxyFactoryBean` uses the `RemoteInvocationFactory` (`AufrufkontextToRemoteInvocationFactory`) by default. 

 However, if you use Spring's own implementation, then this factory must be set.

File: isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc (hunk 2 above)
Comment (FabianPerl) [reply isyfact-standards-doc/src/docs/antora/modules/changelog/pages/migrationsleitfaden.adoc:44]: Scenario A: In the end, `AufrufKontextTo` should no longer be used in IF3 (deprecated),  the SecurityContext should always be used via isy-security. The [attached diagram](https://github.com/IsyFact/isyfact-standards/pull/50/files#diff-4624b7c4a37044be686d2a904a8e6e6bdf35249b962f05eccfe347945c85b5ef) shows which IF version can handle which objects. To ensure compatibility, an automatic transformation of the objects is performed, so no effort is required here.

Scenario B: This case was explicitly not considered in this ticket.

**Review by kollmana — APPROVED** [review]

**Pull request merged**